akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -454,6 +458,9 @@ private[akka] class RemoteActorRef private[akka] (
 
   @volatile var cachedAssociation: artery.Association = null
 
+  // used by artery to direct messages to a separate stream for large messages
+  @volatile var cachedLargeMessageDestinationFlag: LargeMessageDestinationFlag = null
+
```

Comment:
I'm thinking that it might be written by multiple threads, but they will all arrive at the same result so it doesn't really matter, and after that it will stay the same value.